src/library/scala/Product.scala

```diff
@@ -62,10 +62,5 @@ trait Product extends Any with Equals {
 
   /** An iterator over the names of all the elements of this product.
    */
-  def productElementNames: Iterator[String] = new scala.collection.AbstractIterator[String] {
-    private[this] var c: Int = 0
-    private[this] val cmax = productArity
-    def hasNext = c < cmax
-    def next() = { val result = productElementName(c); c += 1; result }
-  }
+  def productElementNames: Iterator[String] = (0 until productArity).iterator map productElementName
```

Comment:
This will box the integers, maybe its best to leave it as before in case it is used in performance sensitive spots.

Reply:
`Range.iterator.next` needs to return an `Integer` as `Iterator` is not specialized.

Reply:
If your version had been submitted as the original version it would have been accepted, I don't think this is super likely to be called from performance critical code.

Reply:
@retronym So it should be the following? `(0 until productArity).map(productElementName).iterator`

Reply:
@retronym the original hasn't shipped (outside of `pre` binaries). Does that change your stance on accepting the change?